test-utils/src/test_dir.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Tools to create fresh temporary directories for testing on demand
+
+use std::{
+    fs, io,
+    path::{Path, PathBuf},
+    sync::{
+        atomic::{AtomicU32, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
+
+use crypto::random::Rng;
+
+fn make_rng() -> impl Rng {
+    let seed = crate::random::Seed::from_entropy();
```

Comment:
Perhaps this should be printed somewhere

Reply:
It does not seem very useful tbh. This is run when the test machinery is initialized, not inside of the tests so it does not play nice with `--nocapture` (is always printed).